server/src/data/certif.rs

```diff
@@ -982,18 +982,23 @@ impl ShamirRecoveryBriefCertificate {
         timestamp: DateTime,
         user_id: UserID,
         threshold: NonZeroU64,
-        per_recipient_shares: HashMap<UserID, NonZeroU64>,
+        per_recipient_shares: Bound<'_, PyDict>,
     ) -> PyResult<Self> {
         Ok(Self(Arc::new(
             libparsec_types::ShamirRecoveryBriefCertificate {
                 timestamp: timestamp.0,
                 author: author.0,
                 user_id: user_id.0,
                 threshold,
-                per_recipient_shares: per_recipient_shares
-                    .into_iter()
-                    .map(|(k, v)| (k.0, v))
-                    .collect(),
+                per_recipient_shares: {
+                    let mut hash_map = HashMap::with_capacity(per_recipient_shares.len());
+                    for (py_key_any, py_value_any) in &per_recipient_shares {
+                        let py_key = py_key_any.downcast::<UserID>()?;
+                        let py_value: NonZeroU64 = py_value_any.extract()?;
+                        hash_map.insert(py_key.borrow().0, py_value);
+                    }
+                    hash_map
+                },
```

Comment:
Out of curiosity, what was the issue with the previous approach?

Reply:
In the previous approach, we were asking PyO3 to convert a Python `dict` into a `HashMap<UserID, NonZeroU64>`, only to re-convert it right away into a `HashMap<libparsec_types::UserID, NonZeroU64>`